src/imitation/rewards/reward_nets.py

```diff
@@ -390,13 +390,17 @@ def predict_processed(
             done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
             update_stats: Whether to update the running stats of the normalization
                 layer.
+            **kwargs: kwargs passed to base predict_processed call.
 
         Returns:
             Computed normalized rewards of shape `(batch_size,`).
         """
         with networks.evaluating(self):
             # switch to eval mode (affecting normalization, dropout, etc)
-            rew_th = self.base.predict_th(state, action, next_state, done)
```

Comment:
I changed this to improve composability of the wrappers. For example, if we add a `NormalizedRewardNet` on top of an `AddSTDRewardWrapper`, we want both to have an effect on the output of `predict_processed`.

While this will cause a performance hit from moving tenors back and forth from the GPU, I think it is worth it. There is probably a better solution, but it would need a larger refactor #464.

Reply:
Good catch -- correctness/composability trumps efficiency.

This makes me wonder why we return things as `numpy` at all, though. AFAICT, the main reason seems to be "RewardFn demands that" / "`VecEnvRewardWrapper` expects it". We could just add a simple utility method that torchifies arrays, and apply that as the very last step at the callee when needed?